datasync/msgsync/plugin_impl_msgsync.go

```diff
@@ -35,7 +34,7 @@ type PubPlugin struct {
 // to not mix with other plugin fields.
 type Deps struct {
 	local.PluginInfraDeps               // inject
-	Messaging             *kafka.Plugin // inject
+	Messaging             messaging.Mux // inject
```

Comment:
Please, check also the examples (replacement of pointers by interfaces needs to be done also in there).